generated/nidaqmx/_base_interpreter.py

```diff
@@ -0,0 +1,1621 @@
+            max_val, units, resistance_config, current_excit_source,
+            current_excit_val):
+        pass
+
+    @staticmethod
+    def create_watchdog_timer_task(
+            self, device_name, session_name, task, timeout, lines, exp_state,
+            initialization_behavior, new_session_initialized):
+        pass
+
+    def create_watchdog_timer_task_ex(
+            self, device_name, session_name, task, timeout,
+            initialization_behavior, new_session_initialized):
+        pass
+
+    def delete_network_device(self, device_name):
+        pass
+
+    def delete_saved_global_chan(self, channel_name):
+        pass
+
+    def delete_saved_scale(self, scale_name):
+        pass
+
+    def delete_saved_task(self, task_name):
+        pass
+
+    def device_supports_cal(self, device_name, cal_supported):
+        pass
+
+    def disable_ref_trig(self, task):
+        pass
+
+    def disable_start_trig(self, task):
+        pass
+
+    def disconnect_terms(self, source_terminal, destination_terminal):
+        pass
+
+    def export_signal(self, task, signal_id, output_terminal):
+        pass
+
+    def get_ai_chan_cal_cal_date(
+            self, task, channel_name, year, month, day, hour, minute):
+        pass
+
+    def get_ai_chan_cal_exp_date(
+            self, task, channel_name, year, month, day, hour, minute):
+        pass
+
+    def get_analog_power_up_states(
+            self, device_name, channel_name, state, channel_type):
+        pass
+
+    def get_analog_power_up_states_with_output_type(
+            self, channel_names, state_array, channel_type_array, array_size):
+        pass
+
+    def get_arm_start_trig_timestamp_val(self, task, data):
+        pass
+
+    def get_arm_start_trig_trig_when(self, task, data):
+        pass
+
+    def get_auto_configured_cdaq_sync_connections(
+            self, port_list, port_list_size):
+        pass
+
+    def get_buffer_attribute_uint32(self, task, attribute, value):
+        pass
+
+    def get_cal_info_attribute_bool(self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_double(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_string(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_uint32(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_chan_attribute_bool(self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_double(self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_double_array(
+            self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_int32(self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_string(self, task, channel, attribute, value, size):
```

Comment:
Python strings are immutable, so the interpreter should represent string out-parameters as return values. The library interpreter should do the "double call" or "IVI dance". The gRPC interpreter should just return strings.

```
def get_chan_attribute_string(self, task, channel, attribute) -> str:
    pass
```

(It's also possible to box string out-parameters in a list or something, but that's kind of hacky. `ctypes` also boxes strings, using string buffer objects, but we don't want to use `ctypes` for the gRPC interpreter.)

Reply:
Sure. Updated.